DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs

```diff
@@ -109,6 +110,12 @@ protected string RedirectURL
                     redirectURL = UrlUtils.ValidReturnUrl(redirectURL);
                 }
 
+                // Reset redirect url for redirecting to 'Redirect After Login' page
+                if (isValidRedirectAfterLoginUrl)

```

Comment:
I do not believe that this is the proper solution, as with RedirectAfterLogin you don't want to lose the functionality of a ReturnUrl.

@david-poindexter or @hismightiness any thoughts?

Reply:
You're correct, Mitch.  Not retaining `ReturnUrl` would be a breaking change.  There are many workflows on other DNN sites that depend on it.

Reply:
@mitchelsellers @hismightiness 
1. Thanks for you feedback. It will be helpful if you could please provide the scenarios where returnUrl should be retained when RedirectAfterLogin flag is set and we have valid redirectAfterLoginUrl value
2. I believe redirectAfterLoginUrl should be the top priority, Please correct me if I am wrong

Reply:
Based on current functionality I would argue that it is the opposite of your priority.

If you have a return URL, the user is trying to login to a specific secured resource, they should be returned back there.

Reply:
@mitchelsellers I feel that is the root cause of the issue https://github.com/dnnsoftware/Dnn.Platform/issues/3661. Either we should close the issue as "Function as Designed" or we should find alternate solution. 
@mitchelsellers @hismightiness Could you please validate https://github.com/dnnsoftware/Dnn.Platform/issues/3661?

Reply:
I think the main cause is that LOGIN_PATH is hardcoded to /login where it should be read from whichever page is the login page in the settings. The suggested fix in the (I think 3rd) comment on the issue, looked appropriate to me I think.

Basically , getting the path from the settings instead of harcoded "login", then check if we come from there, do the redirectAfterLogin, if not respect the RetrunUrl. I believe that would fix the bug without discarding the returnUrl all together

Reply:
IsRedirectingFromLoginUrl() call not hit in my case. https://github.com/dnnsoftware/Dnn.Platform/pull/4171/files#diff-09d8eba5524d6de9ef99d20e271369edL119 returns false

Reply:
I like your suggestion, @valadas

Reply:
@valadas @mitchelsellers @hismightiness Comments addressed and applied proposed solution from https://github.com/dnnsoftware/Dnn.Platform/issues/3661#issuecomment-608557613